From fa3b8916f258a7802fb9a8ba92aa04b71fffb9de Mon Sep 17 00:00:00 2001
From: Ionut Vicovan <Ionut.Vicovan@nxp.com>
Date: Wed, 31 Aug 2022 18:46:05 +0300
Subject: [PATCH 15/19] s32cc:serdes: Enable mode5 in Linux device tree

Mode5 was until now available only in U-Boot, as a demo.
For Linux, it was forced to mode2, via a "fixup", since
it lacked support.
Now Linux has mode5 as well, so remove the "fixup".

Issue: ALB-9179

Upstream-Status: Pending 

Signed-off-by: Ionut Vicovan <Ionut.Vicovan@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/mach-s32/s32-cc/hwconfig_fixups.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/arch/arm/mach-s32/s32-cc/hwconfig_fixups.c b/arch/arm/mach-s32/s32-cc/hwconfig_fixups.c
index 4377849728..c35d15de8f 100644
--- a/arch/arm/mach-s32/s32-cc/hwconfig_fixups.c
+++ b/arch/arm/mach-s32/s32-cc/hwconfig_fixups.c
@@ -779,12 +779,8 @@ static int set_serdes_mode(struct dts_node *root, int id)
 	}
 
 	mode_num = (u32)mode;
-	if (s32_serdes_is_mode5_enabled_in_hwconfig(id)) {
-		if (root->fdt)
-			mode_num = 2;
-		else
-			mode_num = 5;
-	}
+	if (s32_serdes_is_mode5_enabled_in_hwconfig(id))
+		mode_num = 5;
 
 	ret = node_set_prop_u32(&node, "nxp,sys-mode", mode_num);
 	if (ret)
-- 
2.17.1

